tests/staticFilesServer.test.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable indent */
+/* eslint-disable max-len */
+'use strict';
+
+const axios = require('axios');
+const path = require('path');
+const { faker } = require('@faker-js/faker');
+const fs = require('fs');
+const { Server } = require('http');
+const { createServer } = require('../src/createServer.js');
+
+const PORT = 5701;
+const HOST = `http://localhost:${PORT}`;
+
+function generateRandomCSS() {
+  return `
+    /* Random CSS Generated by Faker */
+
+    .${faker.lorem.word()} {
+      color: ${faker.internet.color()};
+      font-size: ${faker.number.int({
+    min: 12, max: 24,
+  })}px;
+    }
+
+    .${faker.lorem.word()} {
+      background-color: ${faker.internet.color()};
+      border: 2px solid ${faker.internet.color()};
+      padding: ${faker.number.int({
+    min: 5, max: 20,
+  })}px;
+    }
+  `;
+}
+
+describe('Static files server', () => {
+  describe('createServer', () => {
+    describe('basic scenarios', () => {
+      it('should create a server', () => {
+        expect(createServer)
+          .toBeInstanceOf(Function);
+      });
+
+      it('should create an instance of Server', async() => {
+        expect(createServer())
+          .toBeInstanceOf(Server);
+      });
+    });
+
+    describe('Server', () => {
+      let server;
+      const publikFolderPath = path.resolve(__dirname, '../public');
+      const stylesFolderPath = path.resolve(__dirname, '../public/styles');
+      const indexFilePath = path.resolve(__dirname, '../public/index.html');
+      const mainCSSFilePath = path.resolve(__dirname, '../public/styles/main.css');
+      const randomCSSContent = generateRandomCSS();
+
+      beforeEach(() => {
+        server = createServer();
+        server.listen(PORT);
+
+        if (!fs.existsSync(publikFolderPath)) {
+          fs.mkdirSync(publikFolderPath);
+        }
+
+        if (!fs.existsSync(stylesFolderPath)) {
+          fs.mkdirSync(stylesFolderPath);
+        }
+
+        if (!fs.existsSync(indexFilePath)) {
+          fs.writeFileSync(indexFilePath, '<!DOCTYPE html>');
+        }
+
+        if (!fs.existsSync(mainCSSFilePath)) {
+          fs.writeFileSync(mainCSSFilePath, randomCSSContent);
+        }
```

Comment:
Conditional logic in tests is not a really good idea

Reply:
I thought if students create all files and directories, we'll probably have an errors